### ISSUE TITLE
false positive on variable type inference

### ISSUE THREAD

**Issue by k4lizen** [python/mypy#20708]
**Bug Report**
Mypy says:
```
pwndbg/aglib/onegadget.py: note: In function "check_constraint":
pwndbg/aglib/onegadget.py:489: error: Argument 1 to "find" has incompatible type "CheckSatResult"; expected "int | Value | None"  [arg-type]
                    page = pwndbg.aglib.vmmap.find(result)
                                                   ^~~~~~
pwndbg/aglib/onegadget.py:501: error: Argument 1 to "find" has incompatible type "CheckSatResult"; expected "int | Value | None"  [arg-type]
                page = pwndbg.aglib.vmmap.find(result)
                                               ^~~~~~
Found 2 errors in 1 file (checked 420 source files)
```
even though `result` cannot have that type at that point.

**To Reproduce**

```bash
git clone git@github.com:k4lizen/pwndbg.git
cd pwndbg
git checkout 3575cd21a7877f845107f1b6808a115a96eac968
uv run --all-groups --all-extras mypy pwndbg/aglib/onegadget.py
```

**Expected Behavior**

I expect mypy to not warn me about this.

**Actual Behavior**

Mypy warns me about this.

**Your Environment**

- Mypy version used: 1.19.1
- Python version used: 3.12.3

**Comment by hauntsaninja** [comment]
Add an explicit type annotation (e.g. `result: object` or `result: int | Value | None`) at the top of your function or use `--allow-redefinition-new`
